File: python/strategy/DuckSimulator/behavior/QuackBehavior.py

class QuackBehavior:
    def quack(self):
        return

class Quack(QuackBehavior):

    def __init__(self, duck_name):
        self.__duck_name = duck_name

    def quack(self):
        print(self.__duck_name + " is quacking")

class Squeak(QuackBehavior):

    def __init__(self, duck_name):
        self.__duck_name = duck_name

    def quack(self):
        print(self.__duck_name + " is squeaking")

class MuteQuack(QuackBehavior):

    def __init__(self, duck_name):
        self.__duck_name = duck_name

    def quack(self):
        print(self.__duck_name + " is not quacking")

File: ruby/strategy/DuckSimulator/behavior/quack_behavior.rb
class QuackBehavior
  def quack
  end
end

class Quack < QuackBehavior

  def initialize(duck_name)
    @duck_name = duck_name
  end

  def quack
    puts(@duck_name + " is quacking")
  end

end

class Squeak < QuackBehavior

  def initialize(duck_name)
    @duck_name = duck_name
  end

  def quack
    puts(@duck_name + " is squeaking")
  end

end

class MuteQuack < QuackBehavior

  def initialize(duck_name)
    @duck_name = duck_name
  end

  def quack
    puts(@duck_name + " is not quacking")
  end

end